test/library/waits.go

```diff
@@ -71,3 +78,49 @@ func conditionsStatusString(conditions []configv1.ClusterOperatorStatusCondition
 	}
 	return result.String()
 }
+
+func WaitForRouteAdmitted(t *testing.T, client routev1client.RouteV1Interface, name, ns string) (string, error) {
+	var admittedURL string
+
+	t.Logf("waiting for route %s/%s to be admitted", ns, name)
+	err := wait.PollImmediate(time.Second, 2*time.Minute, func() (bool, error) {
+		route, err := client.Routes(ns).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("route.Get(%s/%s) error: %v", ns, name, err)
+			return false, nil
+		}
+		for _, i := range route.Status.Ingress {
+			for _, c := range i.Conditions {
+				if c.Type == routev1.RouteAdmitted && c.Status == "True" {
+					admittedURL = i.Host
+					return true, nil
+				}
+			}
+		}
+
+		return false, nil
+	})
+
+	return admittedURL, err
+}
+
+func WaitForHTTPStatus(t *testing.T, waitDuration time.Duration, client *http.Client, targetURL string, expectedStatus int) error {
+	t.Logf("waiting for HEAD at %q to report %d", targetURL, expectedStatus)
+
+	loggingReductionCounter := 0
+	return wait.PollImmediate(time.Second, waitDuration, func() (bool, error) {
+		resp, err := client.Head(targetURL)
+		if err != nil {
+			return false, nil
+		}
+		if resp.StatusCode == expectedStatus {
+			return true, nil
+		}
+
+		if (loggingReductionCounter % 10) == 0 { // only log failure once in 10 seconds
```

Comment:
Given that there could be a range of potential status other than that expected, is it safe assume this log message will be representative of the intervening failures?

Reply:
I'll improve this to only log whenever status changes